staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -790,6 +790,9 @@ func structToUnstructured(sv, dv reflect.Value) error {
 			if err := toUnstructured(fv, subv); err != nil {
 				return err
 			}
+			if fieldInfo.omitempty && subv.IsNil() {
+				continue
```

Comment:
Looks like this isnt' right. json.Marshal +Unmarshal will set `creationTimestamp: nil` and we always need to ensure that it satisfies json marshalling + unmarshalling.

https://github.com/kubernetes/kubernetes/blob/39e341c11f9a18d7fcbf78e4dce6801131e2ba38/staging/src/k8s.io/apimachinery/pkg/runtime/converter.go#L425

This will fail the already existing round trip test for unstructured:

https://github.com/kubernetes/kubernetes/blob/39e341c11f9a18d7fcbf78e4dce6801131e2ba38/pkg/api/testing/unstructured_test.go#L120

Interesting to note that we also have a round trip test specifically for creationTimestamp but it just checks if conversion is possible and not that the round trip values matches.

https://github.com/kubernetes/kubernetes/blob/39e341c11f9a18d7fcbf78e4dce6801131e2ba38/pkg/api/testing/unstructured_test.go#L137

Reply:
What should be the intended behaviour here? Should `creationTimestamp` be present in the map or not? :thinking:

Reply:
> What should be the intended behaviour here? Should creationTimestamp be present in the map or not? thinking

answering myself: from the docs

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

`omitempty` has no meaning for a struct, so `creationTimestamp` should be present in the map.  It is weird that we set the value as nil but in any case, it should be present in the unstructured object/map.

Reply:
shouldn't `omitempty` be removed from `CreateTimestamp` then? maybe via another pull?

Reply:
> shouldn't omitempty be removed from CreateTimestamp then? maybe in another pull? it

yeah, I think we should remove it too. But I thought it was suited for another PR (since it would involved schema changes too afaik and would be a breaking change).

Also, wanted to get feedback on the current changes before removing it. :)